### PR TITLE
Add codeowner files

### PR DESCRIPTION
**Pull request by TravisEz13**
add codeowner file to automatically assign reviewers and to describe who owns areas.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+src/System.Management.Automation/engine/debugger/  @BrucePay @steveL-MSFT @dantraMSFT @PaulHigin
+
+# Area: help
+src/System.Management.Automation/help              @steveL-MSFT @Francisco-Gamino @adityapatwardhan
+
+# Area: Intellisense
+# @daxian-dbw @lzybkr @steveL-MSFT @charub
+
+# Area: Language
+# daxian-dbw @vors @lzybkr @BrucePay @steveL-MSFT
```
Comment (iSazonov) [inline .github/CODEOWNERS]: daxian-dbw -> `@daxian-dbw`

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# https://help.github.com/articles/about-codeowners/
+
+# Areas are not limited to the filters defined in this file
+# First Lets start with areas with no filters or paths
+
+# Area: Performance
+# @lzybkr @steveL-MSFT @adityapatwardhan
```
Comment (iSazonov) [inline .github/CODEOWNERS]: `@steveL-MSFT` -> `@SteveL-MSFT`

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+# Area: Portability
+# @BrucePay @JamesWTruher @steveL-MSFT
+
+# Area: Security
+# @travisez13 @leeholmes @steveL-MSFT @PaulHigin @chunqingchen
```
Comment (iSazonov) [inline .github/CODEOWNERS]: `@travisez13` -> `@TravisEz13`

**Review by iSazonov — COMMENTED** [review]
Nicks should be fixed in all file.

**Comment by TravisEz13** [comment]
I addressed @iSazonov 's feedback

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+# First Lets start with areas with no filters or paths
+
+# Area: Performance
+# @lzybkr @SteveL-MSFT @adityapatwardhan
+
+# Area: Portability
+# @BrucePay @JamesWTruher @SteveL-MSFT
+
+# Area: Security
+# @TravisEz13 @leeholmes @SteveL-MSFT @PaulHigin @chunqingchen
```
Comment (SteveL-MSFT) [inline .github/CODEOWNERS]: You should remove Lee and Chunqing

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]
You should remove my name from everywhere except demos.

**Comment by TravisEz13** [comment]
@SteveL-MSFT  Updated

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+# Area: engine
+src/System.Management.Automation/engine            @daxian-dbw @vors @lzybkr @BrucePay
+
+# Area: Debugging
+# Must be below engine to override
+src/System.Management.Automation/engine/debugger/  @BrucePay@dantraMSFT @PaulHigin
+
+# Area: help
+src/System.Management.Automation/help              @@Francisco-Gamino @adityapatwardhan
```
Comment (adityapatwardhan) [inline .github/CODEOWNERS]: @@Francisco-Gamino -> @Francisco-Gamino

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+# Area: DSC
+src/System.Management.Automation/DscSupport        @TravisEz13
+
+# Area: engine
+src/System.Management.Automation/engine            @daxian-dbw @vors @lzybkr @BrucePay
+
+# Area: Debugging
+# Must be below engine to override
+src/System.Management.Automation/engine/debugger/  @BrucePay@dantraMSFT @PaulHigin
```
Comment (adityapatwardhan) [inline .github/CODEOWNERS]: Space between @BrucePay  and @dantraMSFT

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS [hunk 7]
```diff
@@ -4,74 +4,74 @@
...
-# @lzybkr @SteveL-MSFT @adityapatwardhan
+# @lzybkr @adityapatwardhan
 
 # Area: Portability
-# @BrucePay @JamesWTruher @SteveL-MSFT
+# @BrucePay @JamesWTruher
 
 # Area: Security
-# @TravisEz13 @leeholmes @SteveL-MSFT @PaulHigin @chunqingchen
+# @TravisEz13 @leeholmes @PaulHigin @chunqingchen
```
Comment (SteveL-MSFT) [inline .github/CODEOWNERS]: remove leeholmes, he isn't generally available so Paul, Travis, or I will pull him in when needed.  Remove chunqing

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 7 above)
Comment (TravisEz13) [inline .github/CODEOWNERS]: #Resolved

File: .github/CODEOWNERS (hunk 5 above)
Comment (TravisEz13) [inline .github/CODEOWNERS]: #Resolved

File: .github/CODEOWNERS (hunk 6 above)
Comment (TravisEz13) [inline .github/CODEOWNERS]: Resolved

File: .github/CODEOWNERS [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+src/Microsoft.PowerShell.Commands.Utility/         @JamesWTruher @dantraMSFT @PaulHigin
+
+# Area: Console
+src/Microsoft.PowerShell.ConsoleHost/              @daxian-dbw @lzybkr @anmenaga
+
+# Area: Demo
+demos/                                             @joeyaiello @SteveL-MSFT @HemantMahawar
+
+# Area: DSC
+src/System.Management.Automation/DscSupport        @TravisEz13
```
Comment (dantraMSFT) [inline .github/CODEOWNERS]: You can add dantraMSFT to DSC as well.

Comment (TravisEz13) [inline .github/CODEOWNERS]: Added

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by iSazonov** [comment]
Travis CI temporary failed - I restarted it.

**Review by adityapatwardhan — APPROVED** [review]
LGTM

**Review by daxian-dbw — APPROVED** [review]

**Comment by daxian-dbw** [comment]
> +test/                                              `@JamesWTruher` `@TravisEz13` `@adityapatwardhan`

Just realized that this cause Jim, Travis and Aditya to be dragged into almost every PR review.
Maybe we shouldn't specify 'test/' in the codeowner file.

**Comment by TravisEz13** [comment]
Rules later in the file are supposed to override previous rules.  Let's see how it actually works before we change it.

**Comment by TravisEz13** [comment]
But looking at least one PR it looks like it takes the aggregate of what it finds from the rules.  Do you want me to submit a PR removing the area-test rule?

**Pull request merged**